ac/ac-timedQuiz/src/ActivityRunner.jsx

```diff
@@ -0,0 +1,209 @@
+  )
+    ? shuffle(questionsWithIndex(props))
+    : questionsWithIndex(props);
+  return questions;
+};
+
+const generateExample = (q, progress) => {
+  const curQuestion = q[progress % q.length];
+
+  const startTime = Date.now();
+  return {
+    curQuestion,
+    startTime
+  };
+};
+
+let noAnswerTimeout;
+let delayTimeout;
+
+const Guidelines = ({ start, guidelines, name }) => (
+  <React.Fragment>
+    <div style={styles.text}>Welcome {name}!</div>
+    <div style={styles.guidelines}>{guidelines}</div>
+    <div style={{ ...styles.commands, width: '120px' }}>
+      <Button style={{ ...styles.button, width: '100%' }} onClick={start}>
+        {'Start'}
+      </Button>
+    </div>
+  </React.Fragment>
+);
+
+const CountDownTimer = TimedComponent(({ timeNow, length, start }) => {
+  const timeLeft = Math.ceil((length - Math.ceil(timeNow - start)) / 1000);
+  return <div style={styles.text}>{timeLeft + ' s'}</div>;
+}, 100);
+
+const Delay = ({ next, delay }) => {
+  clearTimeout(delayTimeout);
+  delayTimeout = setTimeout(next, delay);
+  return (
+    <React.Fragment>
+      <div style={styles.text}>Waiting for next question</div>
+      <CountDownTimer start={Date.now()} length={delay} />
+    </React.Fragment>
+  );
+};
+
+const Question = props => {
+  const { setQuestion, question, logger, data, dataFn, activityData } = props;
+  const { curQuestion, startTime } = question;
+
+  const answers = ['answers', 'both'].includes(activityData.config.shuffle)
+    ? shuffle(curQuestion[0].answers)
+    : curQuestion[0].answers;
+
+  const onClick = answer => () => {
+    clearTimeout(noAnswerTimeout);
+    // Logs the question and answer provided
+    const answerTime = Date.now();
+    // Increases the progress and logs the new progress
+    dataFn.numIncr(1, 'progress');
+    // Increases the score and logs the new score
+    const isCorrectAnswer =
+      answer === undefined ||
+      answer.isCorrect === undefined ||
+      !answer.isCorrect
+        ? 0
+        : 1;
+    const timeIncr = Date.now() - startTime;
+    const value = [data.score + isCorrectAnswer, -(data.time + timeIncr)];
+    logger([
+      { type: 'answer', payload: { ...question, answer, answerTime } },
+      {
+        type: 'progress',
+        value: (data.progress + 1) / activityData.config.questions.length
+      },
+      { type: 'score', value }
+    ]);
+    dataFn.numIncr(isCorrectAnswer, 'score');
+    dataFn.numIncr(timeIncr, 'time');
+    // Goes on to next question
+    setQuestion('waiting');
+  };
+
+  clearTimeout(noAnswerTimeout);
+  noAnswerTimeout = setTimeout(onClick(undefined), activityData.config.maxTime);
+
+  return (
+    <React.Fragment>
+      <div style={styles.text}>
+        {curQuestion[0].question.split('\n').map(x => <p key={x}>{x}</p>)}
+      </div>
+      <div style={styles.commands}>
+        {answers.map(option => {
+          const key = curQuestion[1] + option.choice;
+          const buttonA = (
+            <Button
```

Comment:
Why don't you directly return Button instead of first assigning it to a variable?

Reply:
fixed